packages/mock-block-dock/src/use-mock-block-props/use-mock-datastore.ts

```diff
@@ -818,9 +840,110 @@ export const useMockDatastore = (
     [createEntity, readonly],
   );
 
+  const updateStore = (newExternalBlockEntity: Entity) => {
+    const newEntityTypes = [...entityTypes];
+    const entityTypeId =
+      newExternalBlockEntity?.entityTypeId ?? defaultBlockEntityTypeId;
+    const entityId = newExternalBlockEntity?.entityId ?? defaultBlockEntityId;
+
+    const entityTypeInStore = newEntityTypes.find(
+      (entityType) => entityType.entityTypeId === entityTypeId,
+    );
+
+    if (entityTypeInStore) {
+      // If block schema changed, update entity type
+      if (JSON.stringify(blockSchema) !== JSON.stringify(prevBlockSchema)) {
+        Object.assign(entityTypeInStore.schema, blockSchema ?? {});
+        setEntityTypes(newEntityTypes);
+      }
+    } else {
+      // Create a new entity type if the block's entityTypeId isn't in datastore
+      const newBlockEntityType: EntityType = {
+        entityTypeId,
+        schema: {
+          title: "BlockType",
+          type: "object",
+          $schema: "https://json-schema.org/draft/2019-09/schema",
+          $id: `http://localhost/${entityTypeId}`,
+          ...(blockSchema ?? {}),
+        },
+      };
+      setEntityTypes([newBlockEntityType, ...entityTypes]);
+    }
+
+    const newEntities = [...entities];
+    const entityInStore = newEntities.find(
+      (entity) =>
+        entity.entityId === newExternalBlockEntity.entityId &&
+        entity.entityTypeId === newExternalBlockEntity.entityTypeId,
+    );
+
+    if (entityInStore) {
+      // if block entity exists in store and there was no change, do nothing
+      if (
+        JSON.stringify(entityInStore.properties) ===
+        JSON.stringify(newExternalBlockEntity.properties)
+      ) {
+        return;
+      }
+
+      // If the block entity already exists in the store, and
+      // was changed, update what was changed
+      Object.assign(
+        entityInStore.properties,
+        newExternalBlockEntity.properties,
+      );
+      setEntities(newEntities);
+      setBlockEntity(entityInStore);
+    } else {
+      const newBlockEntity = {
+        entityId,
+        entityTypeId,
+        properties: {},
+      };
+
+      if (blockEntity && Object.keys(blockEntity).length > 0) {
+        Object.assign(newBlockEntity.properties, blockEntity.properties);
+      }
+
+      setEntities([newBlockEntity, ...entities]);
+      setBlockEntity(newBlockEntity);
+    }
+  };
+
+  // Update datastore if block isn't in datastore
+  if (
+    !entities.find(
+      (entity) =>
+        entity.entityId === blockEntity.entityId &&
+        entity.entityTypeId === blockEntity.entityTypeId,
+    ) &&
+    [blockEntity.entityId, blockEntity.entityTypeId].every((key) => !!key)
+  ) {
+    updateStore(blockEntity);
+  }
```

Comment:
This ensures we update datastore to reflect accordingly if the block entity isn't present there